tap_github/streams.py

```diff
@@ -375,3 +375,112 @@ def post_process(self, row: dict, context: Optional[dict] = None) -> dict:
             ),
         ),
     ).to_dict()
+
+
+class CommitsStream(GitHubStream):
+    """
+    Defines the 'Commits' stream.
+    The stream is fetched per repository to maximize optimize for API quota
+    usage.
+    """
+
+    name = "commits"
+    path = "/repos/{org}/{repo}/commits"
+    primary_keys = ["node_id"]
+    replication_key = "commit_timestamp"
+    parent_stream_type = RepositoryStream
+    state_partitioning_keys = ["repo", "org"]
+    ignore_parent_replication_key = True
+
+    def get_url_params(
+        self, context: Optional[dict], next_page_token: Optional[Any]
+    ) -> Dict[str, Any]:
+        """Return a dictionary of values to be used in URL parameterization."""
+        params = super().get_url_params(context, next_page_token)
+        if self.replication_key:
+            since = self.get_starting_timestamp(context)
+            if since:
+                params["since"] = since
+        return params
```

Comment:
I'm not confident in this block. Couple of things...

First, I remember recently we found a bug where `get_starting_timestamp()` was sliding along with state updates. TODO on me (@aaronsteers) to check on that and report back on how this would be affected.

Second, this looks very similar and perhaps redundant with what is already happening in the super. Can we remove?

https://github.com/MeltanoLabs/tap-github/blob/main/tap_github/client.py#L59-L72

Reply:
I think I copy/pasted this from `IssueCommentsStream` as the code is exactly the same. If it is redundant here, I guess we can remove it there as well?

Reply:
Yes, I think that's right. I don't see anything happening here that isn't happening in the super() method.